UI/window-basic-media-controls.cpp

```diff
@@ -0,0 +1,239 @@
+	: QDialog (parent),
+	  item    (item_)
+{
+	if (item)
+		source = obs_sceneitem_get_source(item);
+
+	isPlaying = false;
+
+	QString sourceName = QString::fromUtf8(obs_source_get_name(source));
+	setWindowTitle(QTStr("MediaControls") + " - " + sourceName);
+
+	QVBoxLayout *vLayout = new QVBoxLayout();
+	QHBoxLayout *layout = new QHBoxLayout();
+	QHBoxLayout *topLayout = new QHBoxLayout();
+
+	timerLabel = new QLabel();
+	durationLabel = new QLabel();
+
+	playPauseButton = new QPushButton();
+	QPushButton *stopButton = new QPushButton();
+	stopButton->setProperty("themeID", "stopIcon");
+	QPushButton *nextButton = new QPushButton();
+	nextButton->setProperty("themeID", "nextIcon");
+	QPushButton *previousButton = new QPushButton();
+	previousButton->setProperty("themeID", "previousIcon");
+
+	playPauseButton->setMaximumSize(40, 40);
+	stopButton->setMaximumSize(40, 40);
+	nextButton->setMaximumSize(40, 40);
+	previousButton->setMaximumSize(40, 40);
+
+	slider = new QSlider(Qt::Horizontal);
+	slider->setMinimum(0);
+	slider->setMaximum(100);
+
+	timer = new QTimer();
+
+	topLayout->addWidget(timerLabel);
+	topLayout->addWidget(slider);
+	topLayout->addWidget(durationLabel);
+
+	layout->addStretch();
+	layout->addWidget(previousButton);
+	layout->addWidget(playPauseButton);
+	layout->addWidget(stopButton);
+	layout->addWidget(nextButton);
+	layout->addStretch();
+
+	vLayout->addLayout(topLayout);
+	vLayout->addLayout(layout);
+	vLayout->addStretch();
+
+	setLayout(vLayout);
+
+	connect(playPauseButton, SIGNAL(clicked()), this,
+			SLOT(MediaPlayPause()));
+	connect(stopButton, SIGNAL(clicked()), this,
+			SLOT(MediaStop()));
+	connect(nextButton, SIGNAL(clicked()), this,
+			SLOT(MediaNext()));
+	connect(previousButton, SIGNAL(clicked()), this,
+			SLOT(MediaPrevious()));
+	connect(slider, SIGNAL(sliderMoved(int)), this,
+			SLOT(MediaSliderChanged(int)));
+	connect(slider, SIGNAL(sliderReleased()), this,
+			SLOT(MediaSliderReleased()));
+	connect(timer, SIGNAL(timeout()), this,
+			SLOT(Update()));
+
+	SetSliderPosition();
+	SetPlayPauseIcon();
+	SetDurationLabel();
+	SetTimerLabel();
+
+	resize(450, 50);
+	show();
+
+	StartTimer();
+}
+
+OBSBasicMediaControls::~OBSBasicMediaControls()
+{
+	deleteLater();
+}
+
+uint32_t OBSBasicMediaControls::GetMediaTime()
+{
+	return obs_media_get_time(source) / 1000;
+}
+
+uint32_t OBSBasicMediaControls::GetMediaDuration()
+{
+	return obs_media_get_duration(source) / 1000;
+}
+
+void OBSBasicMediaControls::SetSliderPosition()
+{
+	float sliderPosition = ((float)GetMediaTime() /
+					(float)GetMediaDuration()) * 100.0f;
+
+	slider->setValue((int)sliderPosition);
+	SetTimerLabel();
+}
+
+void OBSBasicMediaControls::SetTimerLabel()
+{
+	timerLabel->setText(FormatSeconds(GetMediaTime()));
+}
+
+void OBSBasicMediaControls::SetDurationLabel()
+{
+	durationLabel->setText(FormatSeconds(GetMediaDuration()));
+}
+
+void OBSBasicMediaControls::MediaSeekTo(int value)
+{
+	float seconds = ((((float)value / 100.0f)) *
```

Comment:
There are extra parentheses here.  It should be:
```
	float seconds = ((float)value / 100.0f) *
			obs_media_get_duration(source);
```

You might even be able to lose the first pair of grouping parentheses since the equation _should_ be evaluated left-to-right, which would allow this to be one line.  Up to you on that last part though.